File: Task-12/Task 12.java
import java.util.*;

class Codechef
{
	public static void main (String[] args) throws java.lang.Exception
	{
	    Scanner scan = new Scanner(System.in);
        int t = scan.nextInt();
        while(t-- > 0)
		    {
		    double b = scan.nextDouble(), c = scan.nextDouble();
		    double LL = 0.0, HL = Math.PI/2, x = (HL + LL)/2, LHS = (2*x + b) * Math.tan(x), RHS = (x*x + b*x + c);
		    while(Math.abs(LHS - RHS) > Math.pow(10, -6))
		    {
		        if(LHS >= RHS)
		        {
		            HL = x;
		        }
		        else
		        {
		            LL = x;
		        }
		        x = (HL + LL)/2;
		        LHS = (2*x + b) * Math.tan(x);
		        RHS = (x*x + b*x + c);
		    }
		    double Min = (x*x + b*x + c)/Math.sin(x);
		    System.out.println(Min);
		}
        scan.close();
	}
}

File: Task-12/Task 12.py
import math

b, c = 0, 0
def f(x):
    return (x*x + b*x + c) / math.sin(x)
def m(x): # m = slope of f(x)
    return (2*x + b) * math.sin(x) - (x*x + b*x + c) * math.cos(x)

for t in range (int(input())):
    b, c = map(float, input().split())
    LL, HL = 0.0, math.pi/2
    x = (LL + HL)/2
    while (abs(m(x)) > math.pow(10, -6)):
        if (m(x) > 0):
            HL = x
            x = (LL + HL)/2
        else:
            LL = x
            x = (LL + HL)/2
    print(f(x))